openedx/openedx-platform
# [BD-14] [SE-4587] [DRAFT] Backport Content Libraries LTI 1.3 code to Lilac

## Description

This PR ports and rebases all the changes to support LTI 1.3. on Blockstore-based Content Libraries.  It should contain the same changes already provided by the related PRs, as well as any additional changes related to making it work in the Lilac release.

At this moment, the changes are:

1. https://github.com/edx/edx-platform/pull/27411 - Core LTI 1.3. support code.
1. https://github.com/edx/edx-platform/pull/27715 - Management command to import Courseware blocks (modulestore) into Blockstore through content libraries.
2. https://github.com/edx/frontend-app-library-authoring/pull/22 - MFE changes to allow customers to copy the LTI launch URLs.
3. https://github.com/edx/edx-platform/pull/27712 - Backed changes to support the above.

## Testing instructions

Same as https://github.com/edx/edx-platform/pull/27411 and their companions, except:

1. You should checkout this PR's branch at `edx/edx-platform` repo in your devstack.

## Deadline

None.

## Other information

None.

## Reviewers


- - -
**Settings**
```yaml
EDXAPP_FEATURES:
  ENABLE_COMBINED_LOGIN_REGISTRATION: true

COMMON_ECOMMERCE_BASE_URL: '{{ EDXAPP_ECOMMERCE_PUBLIC_URL_ROOT }}'
EDXAPP_CELERY_BROKER_TRANSPORT: amqp

EDXAPP_VIDEO_IMAGE_SETTINGS:
  VIDEO_IMAGE_MAX_BYTES: 2097152
  VIDEO_IMAGE_MIN_BYTES: 2048

  STORAGE_KWARGS:
    bucket: '{{ EDXAPP_AWS_STORAGE_BUCKET_NAME }}'
    location: video-images/

EDXAPP_VIDEO_TRANSCRIPTS_SETTINGS:
  VIDEO_TRANSCRIPTS_MAX_BYTES: 3145728
  STORAGE_KWARGS:
    bucket: '{{ EDXAPP_AWS_STORAGE_BUCKET_NAME }}'
    location: video-transcripts/

EDXAPP_FEATURES_EXTRA:
  ENABLE_LIBRARY_AUTHORING_MICROFRONTEND: true
  ENABLE_CORS_HEADERS: true

TRUSTED_DOMAINS:
  - 'blockstore.{{ EDXAPP_LMS_BASE }}'
  - 'library-authoring.{{ EDXAPP_LMS_BASE }}'

EDXAPP_ENABLE_CORS_HEADERS: yes
EDXAPP_ENABLE_CROSS_DOMAIN_CSRF_COOKIE: yes
EDXAPP_CORS_ORIGIN_WHITELIST: '{{ TRUSTED_DOMAINS }}'
EDXAPP_LOGIN_REDIRECT_WHITELIST: '{{ TRUSTED_DOMAINS }}'
EDXAPP_CSRF_TRUSTED_ORIGINS: '{{ TRUSTED_DOMAINS }}'
EDXAPP_LIBRARY_AUTHORING_MICROFRONTEND_URL: 'https://library-authoring.{{ EDXAPP_LMS_BASE }}'
MFE_DEPLOY_STANDALONE_NGINX: yes

MFES:
  - name: library-authoring
    repo: frontend-app-library-authoring
    env_extra:
      STUDIO_BASE_URL: https://{{ EDXAPP_CMS_BASE }}

## BLOCKSTORE
## ============

# Do not run blockstore, use Gabor's instance.
SANDBOX_ENABLE_BLOCKSTORE: false
EDXAPP_BLOCKSTORE_API_URL: https://blockstore.gabor-boros.opencraft.hosting/api/v1/

EDXAPP_LMS_ENV_EXTRA:
    BLOCKSTORE_API_AUTH_TOKEN: 19a2e55d-c4fa-4cd2-8f37-d15d01cedb10

EDXAPP_CMS_ENV_EXTRA:
    BLOCKSTORE_API_AUTH_TOKEN: 19a2e55d-c4fa-4cd2-8f37-d15d01cedb10
```
